qiskit/pulse/channels/pulse_channel_spec.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Pulse channel wrapper object for the system.
+"""
+from typing import List
+
+from qiskit.pulse.exceptions import PulseError
+from .channels import AcquireChannel, MemorySlot, RegisterSlot
+from .pulse_channels import DriveChannel, ControlChannel, MeasureChannel
+from .qubit import Qubit
+from .system_topology import SystemTopology
+
+
+class PulseChannelSpec:
+    """A helper class to support assembling channel objects and their mapping to qubits.
+    This class can be initialized with two methods shown as below.
+
+    1. When you have BaseBackend object of the target quantum device.
+        ```python
+        system = PulseChannelSpec.from_device(backend)
+        ```
+
+    2. When you know the number of elements constituting the target quantum device.
```

Comment:
```suggestion
    2. By specifying the number of pulse elements constituting the target quantum system:
```

Reply:
I have a concern about using [quantum system](https://en.wikipedia.org/wiki/Quantum_system) to denote a system consisting of quantum chip + cryosystem + rte. This may remind us of a term in quantum physics. I think _quantum computing system_ sounds more suitable.

Reply:
yes, I agree. Your suggestion makes sense.

Reply:
thanks, I updated.